db_whitebox_test.go

```diff
@@ -112,6 +120,168 @@ func TestMethodPage(t *testing.T) {
+		for cErr := range ch {
+			cErrs = append(cErrs, cErr)
+		}
+		require.Greater(t, len(cErrs), 0)
+
+		t.Log("Check valid pages.")
+		cErrs = cErrs[:0]
+		for _, pgId := range validPageIds {
+			ch = make(chan error)
+			chkConfig.pageId = uint(pgId)
+			go func() {
+				defer close(ch)
+				tx.check(chkConfig, ch)
+			}()
+
+			for cErr := range ch {
+				cErrs = append(cErrs, cErr)
+			}
+			require.Equal(t, 0, len(cErrs))
+		}
+		return nil
+	})
+	require.NoError(t, vErr)
+}
+
+// corruptLeafPage write an invalid leafPageElement into the victim page.
+func corruptLeafPage(t testing.TB, db *DB, pageSize int, expectPanic bool) (victimPageId common.Pgid, validPageIds []common.Pgid) {
+	t.Helper()
+
+	victimPageId, validPageIds = findVictimPageId(t, db)
+
+	victimPage, victimBuf, err := guts_cli.ReadPage(db.Path(), uint64(victimPageId))
+	require.NoError(t, err)
+	require.True(t, victimPage.IsLeafPage())
+	require.True(t, victimPage.Count() > 0)
+
+	// Dumping random bytes in victim page for corruption.
+	copy(victimBuf[32:], generateCorruptionBytes(t, pageSize, expectPanic))
+	// Write the corrupt page to db file.
+	err = guts_cli.WritePage(db.Path(), victimBuf)
+	require.NoError(t, err)
+
+	return victimPageId, validPageIds
+}
+
+// findVictimPageId finds all the leaf pages of a bucket and picks a random leaf page as a victim to be corrupted.
+func findVictimPageId(t testing.TB, db *DB) (victimPageId common.Pgid, validPageIds []common.Pgid) {
+	t.Helper()
+	// Read DB's RootPage.
+	rootPageId, _, err := guts_cli.GetRootPage(db.Path())
+	require.NoError(t, err)
+	rootPage, _, err := guts_cli.ReadPage(db.Path(), uint64(rootPageId))
+	require.NoError(t, err)
+	require.True(t, rootPage.IsLeafPage())
+	require.Equal(t, 1, len(rootPage.LeafPageElements()))
+	// Find Bucket's RootPage.
+	lpe := rootPage.LeafPageElement(uint16(0))
+	require.Equal(t, uint32(common.BranchPageFlag), lpe.Flags())
+	k := lpe.Key()
+	require.Equal(t, "testBucket", string(k))
+	bucketRootPageId := lpe.Bucket().RootPage()
+	// Read Bucket's RootPage.
+	bucketRootPage, _, err := guts_cli.ReadPage(db.Path(), uint64(bucketRootPageId))
+	require.NoError(t, err)
+	require.Equal(t, uint16(common.BranchPageFlag), bucketRootPage.Flags())
+	// Retrieve Bucket's PageIds
+	var bucketPageIds []common.Pgid
+	for _, bpe := range bucketRootPage.BranchPageElements() {
+		bucketPageIds = append(bucketPageIds, bpe.Pgid())
+	}
+
+	randomIdx := rand.Intn(len(bucketPageIds))
+	victimPageId = bucketPageIds[randomIdx]
+	validPageIds = append(bucketPageIds[:randomIdx], bucketPageIds[randomIdx+1:]...)
+	return victimPageId, validPageIds
+}
+
+// generateSampleDataInBucket fill in sample data into given bucket to create the given
+// number of leafPages. To control the number of leafPages, sample data are generated in order.
+func generateSampleDataInBucket(t testing.TB, bk *Bucket, pageSize int, lPages int) {
+	t.Helper()
+
+	maxBytesInPage := int(DefaultFillPercent * float32(pageSize))
```

Comment:
```suggestion
	maxBytesInPage := int(bk.FillPercent * float32(pageSize))
```

Reply:
I wanted to use it, but i remember that you do not want production code in the test 👍🏽